pkg/credentials/file_aws_credentials.go

```diff
@@ -101,7 +146,12 @@ func (p *FileAWSCredentials) Retrieve() (Value, error) {
 
 // IsExpired returns if the shared credentials have expired.
 func (p *FileAWSCredentials) IsExpired() bool {
-	return !p.retrieved
+	if p.expiration.IsZero() {
+		return !p.retrieved
+	}
+
+	now := time.Now()
```

Comment:
you need to add lee-way you can't expire at the last nano-second @albertomurillo look at iam_aws implementation for ideas.

Reply:
Nice, I updated `FileAWSCredentials` struct to make use of credentials.go `Expiry`.

Now that `DefaultExpiryWindow` is being used in two files, what do you think about refactoring it closer to `Expiry` definition in credentials.go ?